internal/services/gateway/api/project.go

```diff
@@ -284,3 +289,49 @@ func (h *ProjectCreateRunHandler) ServeHTTP(w http.ResponseWriter, r *http.Reque
 		h.log.Errorf("err: %+v", err)
 	}
 }
+
+type UpdateProjectRepoInfoHandler struct {
+	log *zap.SugaredLogger
+	ah  *action.ActionHandler
+}
+
+func NewUpdateProjectRepoInfoHandler(logger *zap.Logger, ah *action.ActionHandler) *UpdateProjectRepoInfoHandler {
+	return &UpdateProjectRepoInfoHandler{log: logger.Sugar(), ah: ah}
+}
+
+func (h *UpdateProjectRepoInfoHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	vars := mux.Vars(r)
+	projectRef, err := url.PathUnescape(vars["projectref"])
+	if err != nil {
+		httpError(w, util.NewErrBadRequest(err))
+		return
+	}
+
```

Comment:
A gateway api handler should just fetch the request data and call the associated action.
So all the things done below should be put in a dedicated action instead of calling multiple actions.